sys/include/net/ng_fib/ng_fib.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2014  Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_fib Forwarding Information Base (FIB)
+ * @brief       FIB implementation
+ * @ingroup     net
+ *
+ * @{
+ *
+ * @file        fib.h
+ * @brief       Types and functions for FIB
+ * @author      Martin Landsmann
+ */
+
+#ifndef _FIB_H
+#define _FIB_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Reactive Routing Protocol (RRP) message content to request/reply route discovery
+ */
+typedef struct rrp_address_msg_t {
+    uint8_t *address;      /**< The pointer to the address */
+    uint8_t address_size;  /**< The address size */
+    kernel_pid_t iface_id; /**< The interface used for the next-hop reply.
+                                If it is KERNEL_PID_UNDEF, no next-hop has been found. */
+    uint32_t lifetime;     /**< The lifetime of this next-hop */
+} rrp_address_msg_t;
+
+#define FIB_MSG_RRP_SIGNAL (0x99)              /**< message type for RRP notifications         */
+
+#define FIB_SUCCESS_EXACT_MATCH (1)            /**< return value for successfull exact match   */
+#define FIB_SUCCESS (0)                        /**< return value for success                   */
+#define FIB_ERROR (-1)                         /**< error value for generic failure            */
+#define FIB_ERROR_DESTINATION_UNREACHABLE (-2) /**< error value indicating unreachable dest.   */
+#define FIB_ERROR_INSUFICCIENT_OUT_SIZE (-3)   /**< error value for invalid output buffer size */
```

Comment:
Still the question, if this can be replaced with errno as in the rest of `ng_`. `FIB_ERROR_DESTINATION_UNREACHABLE` can be `-EHOSTUNREACH` and `FIB_ERROR_INSUFICCIENT_OUT_SIZE` can be `-ENOBUFS` on `-EOVERFLOW` depending on the context.